splitgraph/cloud/__init__.py

```diff
@@ -1104,3 +1105,15 @@ def load_all_repositories(self, limit_to: List[str] = None) -> List[Repository]:
             parsed_external = ExternalResponse.from_response(external_r.json())
 
         return make_repositories(parsed_metadata, parsed_external)
+
```

Comment:
IMO this functionality is getting big enough that you can get value from a test that stubs out:

  - `GetTunnelProvisioningToken`
  - upcoming `ProvisionTunnel` GQL
  - rathole `subprocess.run`

and checking the expected GQL calls, the token getting passed between them, their outputs making it correctly into the emitted rathole config etc.

https://github.com/splitgraph/sgr/blob/master/test/splitgraph/commandline/test_cloud.py#L602-L626 has an example of stubbing out GQL calls with `httpretty` and asserting various file outputs using snapshot tests.

Reply:
(the tests are still not here but I think we can solve this separately)